pandas/core/internals/array_manager.py

```diff
@@ -869,6 +869,20 @@ def iset(
             self.arrays[mgr_idx] = value_arr
         return
 
+    def column_setitem(self, loc: int, idx: int | slice | np.ndarray, value) -> None:
+        """
+        Set values ("setitem") into a single column (not setting the full column).
+
+        This is a method on the ArrayManager level, to avoid creating an
+        intermediate Series at the DataFrame level (`s = df[loc]; s[idx] = value`)
+        """
+        arr = self.arrays[loc]
+        # create temporary SingleArrayManager without ref to use setitem implementation
```

Comment:
is the "without ref" here why you're not using .iget?

Reply:
looks like if you used .iget here then the method could be shared between AM/BM?

Reply:
> is the "without ref" here why you're not using .iget?

The "without ref" comment is from copy pasting this from the initial CoW PR https://github.com/pandas-dev/pandas/pull/41878 (so should probably remove that here).

But that is indeed the reason to not use `iget` in that PR.